src/events/SDL_mouse.c

```diff
@@ -680,19 +710,52 @@ SDL_SendPanEvent(
         event.type = SDL_PANEVENT;
         event.pan.windowID = mouse->focus ? mouse->focus->id : 0;
         event.pan.which = mouseID;
-        event.pan.contains_x = contains_x;
-        event.pan.contains_y = contains_y;
-        event.pan.x = precise_x;
-        event.pan.y = precise_y;
-        event.pan.fling = is_fling;
-        event.pan.interrupt = is_interrupt;
-        event.pan.source_type = source_type;
+        event.pan.pantype = SDL_PANEVENTTYPE_FLING;
+        // no union contents written for this event type
+        event.pan.source = SDL_GetPanLastSource();
+        event.pan.axis = axis;
         return SDL_PushEvent(&event) > 0;
```

Comment:
```suggestion
        return SDL_PushEvent(&event);
```
I though I have added this suggestion.